### PR TITLE
Public refactor

### PR DESCRIPTION
**Pull request by Fr3E**

### CODE REVIEW

File: ImagePicker/CellRegistrator.swift [hunk 1]
```diff
@@ -160,19 +114,32 @@ public final class CellRegistrator {
...
     ///
     /// Please note that cell's class must conform to `ImagePickerAssetCell` protocol, otherwise an exception will be thrown.
-    ///
     public func registerNibForAssetItems(_ nib: UINib) {
         assetItemNib = nib
     }
+    
+    func cellIdentifier(forActionItemAt index: Int) -> String? {
+        // First lets check if there is a registered cell at specified index
+        if let index = actionItemNibsData?[index]?.1 ?? actionItemClassesData?[index]?.1 {
```
Comment (radimhalfar) [inline ImagePicker/CellRegistrator.swift:131]: put under guard condition

File: ImagePicker/CellRegistrator.swift [hunk 2]
```diff
@@ -160,19 +114,32 @@ public final class CellRegistrator {
...
+        
+        // If not found globaly registered return nil
+        guard index < Int.max else { return nil }
+        
+        // Lets see if there is a globally registered cell for all indexes
+        return cellIdentifier(forActionItemAt: Int.max)
+    }
+    
+    func cellIdentifier(forAsset type: PHAssetMediaType) -> String? {
+        return assetItemNibsData?[type]?.1 ?? assetItemClassesData?[type]?.1
```
Comment (radimhalfar) [inline ImagePicker/CellRegistrator.swift]: it's not really readable in such format. Can you provide a more readable form of syntax ?

File: ImagePicker/ImagePickerController+Delegate.swift [hunk 3]
```diff
@@ -45,14 +45,14 @@ extension ImagePickerController: ImagePickerDelegateDelegate {
     }
 
     func imagePicker(delegate: ImagePickerDelegate, didEndDisplayingCameraCell cell: CameraCollectionViewCell) {
-        //susped session only if not recording video, otherwise the recording would be stopped.
+        // Susped session only if not recording video, otherwise the recording would be stopped.
```
Comment (radimhalfar) [inline ImagePicker/ImagePickerController+Delegate.swift]: typo

File: ImagePicker/ImagePickerController.swift [hunk 4]
```diff
@@ -1,154 +1,115 @@
...
+    /// added smart album will be used.
+    public var assetsFetchResultBlock: (() -> PHFetchResult<PHAsset>?)?
+
+    /// A collection view that is used for displaying content.
+    public var collectionView: UICollectionView! {
+        return imagePickerView.collectionView
+    }
+    
+    private var collectionViewCoordinator: CollectionViewUpdatesCoordinator!
+    private var imagePickerView: ImagePickerView! {
```
Comment (radimhalfar) [inline ImagePicker/ImagePickerController.swift:42]: not a good idea to use force unwrapping even it is in the framework itself.

**Review by radimhalfar — CHANGES_REQUESTED** [review]

File: ImagePicker/CellRegistrator.swift (hunk 1 above)
Comment (Fr3E) [reply ImagePicker/CellRegistrator.swift:131]: I think its inteded to be as it is. Its some obscure recursion

File: ImagePicker/CellRegistrator.swift (hunk 2 above)
Comment (Fr3E) [reply ImagePicker/CellRegistrator.swift]: Tried, but cant

File: ImagePicker/ImagePickerController.swift (hunk 4 above)
Comment (Fr3E) [reply ImagePicker/ImagePickerController.swift:42]: Yes, but i dont see a way how to recover from this, its better to crash at this line than have many checks when variable is used.

File: ImagePicker/CellRegistrator.swift (hunk 2 above)
Comment (Fr3E) [reply ImagePicker/CellRegistrator.swift]: changed .1 to identifier

**Review by radimhalfar — APPROVED** [review]

**Pull request merged**